src/jit/rangecheck.cpp

```diff
@@ -1050,6 +1050,15 @@ bool RangeCheck::ComputeDoesOverflow(BasicBlock* block, GenTree* expr)
     {
         overflows = DoesPhiOverflow(block, expr);
     }
+    else if (expr->OperGet() == GT_CAST)
+    {
+        GenTreeCast* castExpr = expr->AsCast();
+        if (castExpr->CastToType() == TYP_UBYTE && castExpr->CastFromType() == TYP_INT)
```

Comment:
There doesn't seem to be any requirement that `CastToType` be only `UBYTE`. `USHORT` would also work (though 65536 element tables are probably less common than 256 ones).

Reply:
@mikedn I agree, I probably even have one. in my managed port of https://github.com/lemire/simdjson library. Also, the lib has 4 tables of bytes (Length=256).